ad713x-fmcz/ad713x.c

```diff
@@ -239,15 +245,55 @@ int32_t ad713x_mag_phase_clk_delay(ad713x_dev *dev,
 				   bool clk_delay_en)
 {
 	int32_t ret;
+	int8_t i;
 
-	ret = ad713x_spi_write_mask(dev,
-				    AD713X_REG_DEVICE_CONFIG5,
-				    AD713X_DEV_CONFIG5_MPC_CLKDEL_EN_MSK,
-				    AD713X_DEV_CONFIG4_MPC_CLKDEL_EN_MODE(clk_delay_en));
+	if(clk_delay_en) {
+		for(i = CH3; i >= 0; i--) {
```

Comment:
use just one for and a aux variable for AD713X_MPC_CLKDEL_EN_CH_MODE. 

Then you do this:
if(clk_delay_en)
    temp = DELAY_2_CLOCKS;
else 
    temp = DELAY_NONE;
for(i = CH3; i >= 0; i--) {
			ret = ad713x_spi_write_mask(dev,
						    AD713X_REG_MPC_CONFIG,
						    AD713X_MPC_CLKDEL_EN_CH_MSK(i),
						    AD713X_MPC_CLKDEL_EN_CH_MODE(temp, i));
			if(ret != 0)
				break;
}